use_cases/eluc/app/components/dms/chart.py

```diff
@@ -37,60 +38,61 @@ def register_update_context_chart_callback(self, app):
             :param year: Selected context year.
             :param lat: Selected context lat.
             :param lon: Selected context lon.
-            :return: New figure type selected by chart_type with data context.
+            :return: New figure with data context.
             """
-            context = self.df.loc[year, lat, lon]
-            chart_data = utils.add_nonland(context[constants.LAND_USE_COLS])
-
-            assert chart_type in ("Treemap", "Pie Chart")
+            context = self.df.loc[[year], [lat], [lon]]
+            chart_data = utils.add_nonland(context[constants.LAND_USE_COLS]).iloc[0]
 
-            if chart_type == "Treemap":
-                return self.create_treemap(chart_data, type_context=True, year=year)
-
-            return self.create_pie(chart_data, type_context=True, year=year)
+            return self.create_treemap(chart_data, type_context=True, year=year)
 
     def register_update_presc_chart_callback(self, app):
         """
         Callback that updates prescription chart when prescription sliders are updated or chart type is changed.
         """
         @app.callback(
-            Output("presc-fig", "figure"),
-            Input("chart-select", "value"),
-            Input({"type": "presc-slider", "index": ALL}, "value"),
+            Output("presc-fig", "figure", allow_duplicate=True),
+            Output("alert", "is_open"),
+            Input("update-button", "n_clicks"),
+            [State({"type": "diff-slider", "index": ALL}, "value")],
             State("year-input", "value"),
             State("lat-dropdown", "value"),
             State("lon-dropdown", "value"),
-            prevent_initial_call=True
         )
-        def update_presc_chart(chart_type, sliders, year, lat, lon):
+        def update_presc_chart(n_clicks, sliders, year, lat, lon):
             """
             Updates prescription chart from sliders according to chart type.
-            :param chart_type: String input from chart select dropdown.
+            :param n_clicks: Number of times update button has been clicked.
             :param sliders: Prescribed slider values.
             :param year: Selected context year (also for title of chart).
             :param lat: Selected context lat.
             :param lon: Selected context lon.
-            :return: New chart of type chart_type using presc data.
+            :return: New chart using presc data.
             """
-            # If we have no prescription just return an empty chart
-            if all(slider == 0 for slider in sliders):
-                return self.create_treemap(pd.Series([], dtype=float), type_context=False, year=year)
 
-            presc = pd.Series(sliders, index=constants.RECO_COLS)
+            # If we have no prescription just show the context chart
+            bad_sliders = sum(sliders) < -0.01 or sum(sliders) > 0.01
+            if all(slider == 0 for slider in sliders) or bad_sliders:
```

Comment:
We add a `prescribe + predict` button instead of automatically updating the chart so we don't have to validate that the sliders sum to 100 on the fly